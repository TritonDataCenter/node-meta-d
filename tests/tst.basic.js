var mod_path = require('path');
var mod_testcaselib = require('./testcaselib');

mod_testcaselib.runTestCases({
    'name': mod_path.basename(__filename),
    'fieldtypes': {
	'terra': 'string',
	'celes': 'number'
    },
    'input': {
	'fields': [ 'terra', 'celes' ],
	'metad': {
		'usepragmazone': true,
		'probedesc': [ {
			'probes': [ 'syscall:::entry' ],
			'gather': {
				'celes': {
					'gather': 'vtimestamp',
					'store': 'thread'
				}
			}
		}, {
			'probes': [ 'syscall:::return' ],
			'aggregate': {
				'default': 'count()',
				'terra': 'count()',
				'celes': 'quantize($0)'
			},
			'transforms': {
				'terra': 'zonename',
				'celes': 'timestamp - $0'
			},
			'verify': {
				'celes': '$0'
			}
		}, {
			'probes': [ 'syscall:::return' ],
			'clean': {
				'celes': '$0'
			}
		} ]
	}
    },
    'cases': [ {
	/* simple case */
	'predicate': {},
	'decomposition': []
    }, {
	/*
	 * uses "#pragma D option zone" as well as a zone predicate that gets
	 * combined with the existing predicates
	 */
	'zones': [ 'zone1', 'zone2' ],
	'predicate': { 'gt': [ 'celes', 10 ] },
	'decomposition': []
    }, {
	/* like previous, but too many zones to use the #pragma */
	'zones': [ 'zone1', 'zone2', 'zone3', 'zone4', 'zone5' ],
	'predicate': { 'gt': [ 'celes', 10 ] },
	'decomposition': []
    }, {
	/* decomposition combinations */
	'predicate': {},
	'decomposition': [ 'celes' ]
    }, {
	'predicate': {},
	'decomposition': [ 'terra' ]
    }, {
	'predicate': {},
	'decomposition': [ 'celes', 'terra' ]
    }, {
	/* predicates and decompositions */
	'predicate': { 'gt': [ 'celes', '300' ] },
	'decomposition': [ 'celes' ]
    }, {
	'predicate': { 'gt': [ 'celes', '300' ] },
	'decomposition': []
    }, {
	'predicate': { 'or': [
	    { 'eq': [ 'terra', 'global' ] },
	    { 'eq': [ 'terra', 'adm' ] }
	] },
	'decomposition': []
    } ]
});
