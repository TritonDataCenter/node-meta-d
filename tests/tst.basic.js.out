/*
 * test case 0:
 *     zones = []
 *     predicate = {}
 *     decomps = []
 */
/* start script 0 */
syscall:::return
{
	@ = count();
}

/* end script 0 */

/*
 * test case 1:
 *     zones = ["zone1","zone2"]
 *     predicate = {"gt":["celes",10]}
 *     decomps = []
 */
/* start script 0 */
#pragma D option zone=zone1

syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))) && (((zonename == "zone1") || (zonename == "zone2"))) && ((timestamp - self->celes0) > 10))/{
	@ = count();
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/* start script 1 */
#pragma D option zone=zone2

syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))) && (((zonename == "zone1") || (zonename == "zone2"))) && ((timestamp - self->celes0) > 10))/{
	@ = count();
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 1 */

/*
 * test case 2:
 *     zones = ["zone1","zone2","zone3","zone4","zone5"]
 *     predicate = {"gt":["celes",10]}
 *     decomps = []
 */
/* start script 0 */
syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))) && (((zonename == "zone1") || (zonename == "zone2") || (zonename == "zone3") || (zonename == "zone4") || (zonename == "zone5"))) && ((timestamp - self->celes0) > 10))/{
	@ = count();
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/*
 * test case 3:
 *     zones = []
 *     predicate = {}
 *     decomps = ["celes"]
 */
/* start script 0 */
syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))))/{
	@ = quantize((timestamp - self->celes0));
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/*
 * test case 4:
 *     zones = []
 *     predicate = {}
 *     decomps = ["terra"]
 */
/* start script 0 */
syscall:::return
{
	@[(zonename)] = count();
}

/* end script 0 */

/*
 * test case 5:
 *     zones = []
 *     predicate = {}
 *     decomps = ["celes","terra"]
 */
/* start script 0 */
syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))))/{
	@[(zonename)] = quantize((timestamp - self->celes0));
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/*
 * test case 6:
 *     zones = []
 *     predicate = {"gt":["celes","300"]}
 *     decomps = ["celes"]
 */
/* start script 0 */
syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))) && ((timestamp - self->celes0) > "300"))/{
	@ = quantize((timestamp - self->celes0));
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/*
 * test case 7:
 *     zones = []
 *     predicate = {"gt":["celes","300"]}
 *     decomps = []
 */
/* start script 0 */
syscall:::entry
{
	self->celes0 = vtimestamp;
}

syscall:::return
/((((((self->celes0) != NULL)))) && ((timestamp - self->celes0) > "300"))/{
	@ = count();
}

syscall:::return
{
	(self->celes0) = 0;
}

/* end script 0 */

/*
 * test case 8:
 *     zones = []
 *     predicate = {"or":[{"eq":["terra","global"]},{"eq":["terra","adm"]}]}
 *     decomps = []
 */
/* start script 0 */
syscall:::return
/((((zonename) == "global") || ((zonename) == "adm")))/{
	@ = count();
}

/* end script 0 */

